### PR TITLE
Ui/fix demoting status menu

### PR DESCRIPTION
**Pull request by noelledaley**
# Fix Status Menu
This PR fixes 2 bugs with the status menu:
1. When a Replication DR Primary was demoted to a DR Secondary, the status menu still appeared to show it as a Primary, resulting in errors if a user navigated anywhere. This PR fixes this by hiding all links from the status menu when viewing a secondary, as secondaries cannot make those requests anyway.

2. The status menu did not show the replication type (i.e. "DR" or "Performance"). It simply said "Primary", "Secondary" or "Enable", which made the case where someone might have DR enabled but Performance disabled confusing.

### Viewing a DR Primary with Performance disabled
Replication type (DR or Performance) is displayed and the links to disabled replication modes are greyed out.
dr

### Viewing a DR Secondary 
All inaccessible links are hidden!
dr2

### CODE REVIEW

File: ui/app/templates/partials/status/cluster.hbs [hunk 1]
```diff
@@ -132,22 +135,20 @@
           </li>
         </ul>
       </nav>
-      {{#if (has-permission 'metrics' routeParams='requests')}}
+      {{#if ( and (has-permission 'metrics' routeParams='requests') (not cluster.dr.isSecondary))}}
```
Comment (meirish) [inline ui/app/templates/partials/status/cluster.hbs]: So I was wrong that this doesn't render on sealed pages - HTTP metrics still shows up if the vault is sealed or un-inititialized. I think there should be a check for an authenticated user here too, then it wouldn't show up on init / sealed pages.

Comment (noelledaley) [reply ui/app/templates/partials/status/cluster.hbs]: good call, i added this check in https://github.com/hashicorp/vault/pull/7997/commits/c469895ed63bb5503e8f4be74cfa5cf5124ff4d7. thanks!

File: ui/app/templates/partials/status/cluster.hbs [hunk 2]
```diff
@@ -132,22 +135,20 @@
...
-      {{#if (has-permission 'metrics' routeParams='requests')}}
+      {{#if ( and (has-permission 'metrics' routeParams='requests') (not cluster.dr.isSecondary) auth.currentToken)}}
         <hr />
         <nav class="menu">
           <div class="menu-label">
             Metrics
           </div>
           <ul class="menu-list">
             <li class="action">
-            {{#if activeCluster.unsealed}}
```
Comment (johncowen) [inline ui/app/templates/partials/status/cluster.hbs]: It's hard for me to get a good grasp of this, but! This looks like one of those changes which is basically 'I just wrapped something in an `if` but git thinks I changed the world' 😄  

The only thing that jumps out at me as 'why was this changed?' is this removal of `activeCluster.unsealed` here. I thought I'd just check with you to make sure you meant to remove this just incase.

**Comment by joshuaogle** [comment]
This looks great, much less confusing than what we had. :+1:

One thing I noticed is that it looks a little odd with no icon on the right, especially when you have both modes enabled. Can we add in a checkmark when the item is enabled? 
image

File: ui/app/templates/partials/status/cluster.hbs (hunk 2 above)
Comment (noelledaley) [reply ui/app/templates/partials/status/cluster.hbs]: nice catch! you're right that this is related to matthew's comment. i opted to remove this check and instead [check for an authenticated user](https://github.com/hashicorp/vault/pull/7997/files#diff-2b7fa667fa149e97c987bb886b1934c9R138) since that handles the case where vault is sealed as well. this way we hide the entire "Metrics" navigation item too -- not just the "HTTP Requests" link.
